app/src/main/java/ar/com/wolox/android/example/ui/login/LoginPresenter.java

```diff
@@ -61,6 +68,30 @@ public void onLoginClicked(@NonNull String email, @NonNull String password) {
         return errors;
     }
 
+    private void authenticate(String email, String password) {
+        UserAuthService userAuthService = retrofitService.getService(UserAuthService.class);
+        userAuthService.findUserByEmailAndPassword(email).enqueue(getCallback(email, password));
+    }
+
+    private Callback<List<User>> getCallback(String email, String password) {
+        return new Callback<List<User>>() {
+            @Override
+            public void onResponse(@NonNull Call<List<User>> call, @NonNull Response<List<User>> response) {
+                List<User> users = response.body();
+                if (users != null && !users.isEmpty() && users.get(0).getPassword().equals(password)) {
```

Comment:
you can abstract `users != null && !users.isEmpty() && users.get(0).getPassword().equals(password)` to a new method like `isUserInfoValid` or something like that :)